modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -44,76 +44,145 @@ class Instrument_Manager extends \NDB_Menu_Filter
+            // Don't update the file if it already exists on the back-end.
+            // Instead, inform users that an administrator must install it on
+            // their behalf.
+            // This should only happen for users on a system where automatic
+            // installation is disabled (ie. has no quatUser), as the above
+            // error will return before this one.
+            $file_path = $this->path . "project/instruments/" .
+                        $_FILES['install_file']['name'];
+            if (file_exists($file_path)) {
+                $this->tpl_data['feedback'] = "'$instname.linst' has already "
+                    . "been uploaded. Please contact your administrator to "
+                    . "install the instrument.";
+                http_response_code(409);
+                return;
+            }
+
+            move_uploaded_file($_FILES['install_file']['tmp_name'], $file_path);
+            chmod($file_path, 0644);
+            // Scripts in tools/ often make relative imports, so we must change
+            // our effective directory in order to use them.
+            chdir($this->path . "/tools");
+
+            // Use tools/ script to generate an SQL patch file based on the
+            // structure of the uploaded .linst file.
+            // If no quatUser is configured, automatic installation is not
+            // possible, so this is the last step.
+            $db_config = $config->getSetting('database');
+            exec(
+                'php generate_tables_sql_and_testNames.php < '
+                . escapeshellarg($file_path)
+            );
+            if (!$quat_user_is_configured) {
+                http_response_code(201);
+                return;
+            }
+
+            // Install the instrument by directly sourcing the SQL file
+            // generated by `generate_tables_sql_and_testNames.php` using bash.
+            // If installation is successful, `exec` will complete
+            // silently. Otherwise, it will return the exit code and error
+            // messsage from MySQL. This will be stored in $result and
+            // logged via LorisException.
+            $instrument = \NDB_BVL_Instrument::factory(
+                $instname,
+                '',
+                ''
+            );
+            exec(
+                "mysql".
+                " -h" . escapeshellarg($db_config['host']).
+                " -u" . escapeshellarg($db_config['quatUser']).
+                " -p" . escapeshellarg($db_config['quatPassword']).
+                " " . escapeshellarg($db_config['database']).
+                " < " . $this->path . "project/tables_sql/".
+                escapeshellarg($instrument->table . '.sql'),
+                $output, // $output and $status are created automatically
+                $status  // by `exec` and so need not be declared above.
+            );
+            // In MSQL, an exit code of 0 is a success and 1 means failure
```

Comment:
mysql.

And that's not a MySQL thing, that's a unix standard across all tools.